### ISSUE TITLE
Tor integration question

### ISSUE THREAD

**Issue by DigiAngel** [droe/sslsplit#98]
So...when I'm doing malware analysis I have a machine that acts as a MItM with two NIC's bridged.  Currently using iptables PREROUTE table I can pipe all traffic through a listening tor instance setup with tor-socks.  I also have sslsplit listening as well and can pipe traffic through it.  I'd like to try and do both, but I'm not sure which to do first, or how.  Is there a way to, in layman's terms, "grab all https traffic, put it through sslsplit first, then send it along via tor"?  I've tried a few different ways, but I've never been successful.  Thanks for any hints.

**Comment by droe** [comment]
That's more of a Netfilter (iptables) question than an sslsplit question. Generally speaking you will need to refine your Netfilter rule that causes traffic to go through tor to only match connections coming from the sslsplit binary, and redirect all SSL traffic to sslsplit instead of into tor. How this will look depends a lot on your current setup.

**Comment by DigiAngel** [comment]
Hi Droe, thanks for looking at this.  So here's what I have listening:
tor-socks and tor at 9050 and 9040
sslsplit at 8443

My iptables is below (bridged eth0:eth1):
iptables -t nat -A PREROUTING -m physdev --physdev-in eth1 -p tcp --dport -j REDIRECT to-ports 8443

I can REDIRECT to tor at 9040, OR REDIRECT to sslsplit at 8443, but I can't seem to get tor to pass the traffic to sslsplit, or get sslsplit to pass the traffic to tor.  Does that help explain it a little better?  Thanks again.

**Comment by droe** [comment]
Well, that you cannot pass the traffic from tor to sslsplit is obvious, since tor is not SSL/TLS. SSLsplit does not speak SOCKS for the upstream, so telling sslsplit to talk to tor-socks is not an option either. What you want to do is called transparent torification. You will need to combine transparent torification Netfilter rules with an SSLsplit Netfilter rules setup. If that does not work, you can always separate transparent torification and SSL splitting to separate systems.

**Comment by DigiAngel** [comment]
Thank you....I'll do some digging on transparent torification.
